src/pages/question/panels/EdgePanel.jsx

```diff
@@ -83,15 +112,30 @@ export default function EdgePanel(props) {
       }),
     ).filter((n) => !n.deleted);
 
+  const disablePredicates = !(edge.sourceId && edge.targetId);
+
   // Determine default message for predicate selection component
   let predicateInputMsg = 'Choose optional predicate(s)...';
-  if (!edge.predicatesReady) {
-    predicateInputMsg = 'Loading...';
-  } else if (edge.disablePredicates) {
+  if (disablePredicates) {
     predicateInputMsg = 'Source and/or Target Nodes need to be specified...';
   }
+
+  // Every predicate selected must match at least one
+  // predicate in the filteredPredicateList
+  const isValidPredicate = edge.predicate.every(
+    (p) => filteredPredicateList.some((fp) => p.name === fp.name),
+  );
+
+  const isValid = edge.sourceId && edge.targetId && isValidPredicate;
+
   const disabledSwitch = edge.sourceId === null || edge.targetId === null;
 
+  useEffect(() => {
+  // Update edge in panelStore
+    edge.setIsValid(isValid);
+    edge.setIsValidPredicate(isValidPredicate);
+  }, [edge.sourceId, edge.targetId, edge.predicate]);
```

Comment:
I agree that this is an issue. The problem is that isValid is read directly in useNewQuestionPanel so it needs to be a property of useEdge. We can't move all of the logic for isValid into useEdge because it includes API calls, which require access to alert for error handling (unavailable in hooks). 

The [refactoring proposal](https://github.com/NCATS-Gamma/robokop-rewrite/issues/45) provides a more permanent solution to this issue by requiring a single source of state and moving panel controls within the EdgePanel.